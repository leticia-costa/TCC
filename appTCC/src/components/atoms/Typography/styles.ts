import { StyleSheet } from "react-native";
import colors from "../../../utils/colors";

export const styles = StyleSheet.create({
    default: {
      fontFamily: "Arial",
      fontSize: 14,
      color: colors.black,
    },
    heading: {
      fontFamily: "Helvetica-Bold",
      fontSize: 18,
      color: colors.dark,
    },
    subheading: {
      fontFamily: "Arial",
      fontSize: 16,
      color: colors.black,
    },
  });
  