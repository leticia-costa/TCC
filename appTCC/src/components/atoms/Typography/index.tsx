import React from "react";
import { Text, StyleProp, TextStyle } from "react-native";
import { styles } from "./styles";
import { ColorProp } from "../../../utils/colors";

interface TypographyProps {
  style?: StyleProp<TextStyle>;
  children: React.ReactNode;
  variant?: "default" | "heading" | "subheading";
}

const Typography: React.FC<TypographyProps> = ({
  style,
  children,
  variant = "default",
}) => {
  const textStyles = styles[variant] || styles.default;

  return <Text style={[textStyles, style]}>{children}</Text>;
};

export default Typography;
