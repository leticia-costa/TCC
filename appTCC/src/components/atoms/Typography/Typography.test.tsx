import React from "react";
import renderer from "react-test-renderer";
import Typography from ".";

it("renders correctly", () => {
  const tree = renderer.create(<Typography>Teste</Typography>).toJSON();
  expect(tree).toMatchSnapshot();
});
