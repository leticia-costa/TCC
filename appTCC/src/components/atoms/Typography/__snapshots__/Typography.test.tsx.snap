// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "Arial",
        "fontSize": 14,
      },
      undefined,
    ]
  }
>
  Teste
</Text>
`;
